skglm/estimators.py

```diff
@@ -133,16 +133,26 @@ def _glm_fit(X, y, model, datafit, penalty):
                 "The size of the WeightedL1 penalty weights should be n_features, "
                 "expected %i, got %i." % (X_.shape[1], len(penalty.weights)))
 
+    # TODO Handle multi-task case
     if is_classif:
-        solver = cd_solver  # TODO to be be replaced by an instance of BaseSolver
+        if isinstance(datafit, Logistic):
+            solver = ProxNewton(
+                p0=model.p0, tol=model.tol, fit_intercept=model.fit_intercept,
+                max_iter=model.max_iter, max_epochs=model.max_epochs,
+                verbose=model.verbose)
+        else:
+            solver = AcceleratedCD(
+                fit_intercept=model.fit_intercept, max_iter=model.max_iter,
+                max_epochs=model.max_epochs, p0=model.p0, tol=model.tol,
+                ws_strategy=model.ws_strategy, verbose=model.verbose)
     else:
-        solver = cd_solver if y.ndim == 1 else multitask_bcd_solver
-    # TODO this must be replaced by an instance of BaseSolver being passed
-    # so that arguments are attributes of the `solver` object and arguments
-    # do not need to match across solvers
+        solver = AcceleratedCD(
+                fit_intercept=model.fit_intercept, max_iter=model.max_iter,
+                max_epochs=model.max_epochs, p0=model.p0, tol=model.tol,
+                ws_strategy=model.ws_strategy, verbose=model.verbose)
     # TODO QUESTIONS
     # What about ws_strategy?
-    coefs, p_obj, kkt = solver(
+    coefs, p_obj, kkt = solver.solve(
         X_, y, datafit_jit, penalty_jit, w, Xw, max_iter=model.max_iter,
```

Comment:
all paremeters other X, y, df, pen, w and Xw should be removed and accessed as `solver.XXX`

otherwise you need a huge signature here that has all arguments of all solvers